pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -2107,6 +2106,7 @@ func (r *ClickHouseReader) SetTTL(ctx context.Context,
 					}
 					return
 				}
+				req += fmt.Sprint(" SETTINGS distributed_ddl_task_timeout = 0;")
```

Comment:
Doesn't this succeed immediately, and then its status gets updated to SUCCESS in SQLite, although its status is either pending or failed in the queue?

Reply:
Right now we are checking if TTL query successful by comparing last TTL set by user (stored in SQLite DB) and actual TTL in Clickhouse DB.

Reply:
https://github.com/SigNoz/signoz/blob/1c8626e933c9e2d45605b08c38abac22ec582439/pkg/query-service/app/clickhouseReader/reader.go#L2252-L2269

I see this is used in some places, which relies on the `ttl_status` from SQLite?

Reply:
Yes

Reply:
But in getTTL api we fetching TTL from Clickhouse
https://github.com/SigNoz/signoz/blob/285560944edbce1048c212b7f223991abfa14ffc/pkg/query-service/app/clickhouseReader/reader.go#LL2436C25-L2436C25

Reply:
You are right that user will see success but TTL won't be set and older TTL will be seen.
Should we keep `distributed_ddl_task_timeout` as infinite instead of async, that way user can see correct status

Reply:
I am a little skeptical of using infinite, but that only seems like the option if we don't want to rewrite most of the status part. Can you check how it affects? if you don't see any issue then I think we can go with it.

Reply:
Yeah, updated to make it sync. Tested too.

Reply:
Oh, changes weren't pushed.
Pushed now